pkg/rkeworker/docker.go

```diff
@@ -82,25 +83,37 @@ func runProcess(ctx context.Context, name string, p v3.Process, start, forceRest
 			}
 		}
 	}
-
 	for i := 1; i < len(matchedContainers); i++ {
 		if err := remove(ctx, c, matchedContainers[i].ID); err != nil {
 			return err
 		}
 	}
 
 	if len(matchedContainers) > 0 {
-		if strings.Contains(name, "share-mnt") {
-			inspect, err := c.ContainerInspect(ctx, matchedContainers[0].ID)
-			if err != nil {
-				return err
-			}
+		inspect, err := c.ContainerInspect(ctx, matchedContainers[0].ID)
+		if err != nil {
+			return err
+		}
+
+		// share-mnt does not need to be in running state/does not have to be restarted if it ran successfully
+		if strings.Contains(name, ShareMntContainerName) {
 			if inspect.State != nil && inspect.State.Status == "exited" && inspect.State.ExitCode == 0 {
 				return nil
 			}
 		}
+		// ignore service-sidekick if it is present (other containers just use the volumes)
+		if strings.Contains(name, services.SidekickContainerName) {
```

Comment:
this can create a problem if the service sidekick ever updated, service sidekick is responsible for cloudprovider scripts, and other entry points needed for system components

Reply:
This is caught above in the if changed -> remove, then it continues on creating the new ones. Upgrades have been tested with new rke-tools (2.2.5 -> rancher with this fix included)